service_crategen/src/commands/generate/codegen/mod.rs

```diff
@@ -428,25 +428,32 @@ fn generate_struct<P>(service: &Service,
     }
 
     let attributes = format!("#[derive({})]", derived.join(","));
+    let test_attributes = if derived.iter().any(|&x| x == "Deserialize") && !derived.iter().any(|&x| x == "Serialize") {
```

Comment:
this is the relevant change that enables things that are typically Deserialized and not Serialized by default ( i.e. Result types ) to be Serialized but only under test for the sake of keeping binaries as small as they can be and still enabling better test ergonomics.